tokio/src/sync/mpsc/bounded.rs

```diff
@@ -299,6 +299,28 @@ impl<T> Receiver<T> {
     pub fn poll_recv(&mut self, cx: &mut Context<'_>) -> Poll<Option<T>> {
         self.chan.recv(cx)
     }
+
+    /// Creates a [`Sender`] that can be used to send messages to this receiver.
+    ///
+    ///
+    /// # Examples
```

Comment:
What happens in these cases?

 1. The channel was closed because all existing senders were dropped.
 2. The channel was closed because `Receiver::close` was called. There are still some existing senders.
 3. The channel was closed because `Receiver::close` was called. All senders have been dropped as well, but this happened after the call to `close`.

Reply:
Yeah - I mentioned this in https://github.com/tokio-rs/tokio/pull/3713#issuecomment-821911661 - I don't know if this is feasible to implement given that. 

Any clever ideas? Otherwise I might just close this PR! :(